components/p3a/p3a_config.cc

```diff
@@ -36,15 +37,18 @@ void MaybeReadStringFromCommandLine(base::CommandLine* cmdline,
   }
 }
 
-void MaybeReadURLFromCommandLine(base::CommandLine* cmdline,
-                                 const char* switch_name,
-                                 GURL* config_value) {
+void MaybeReadURLFromCommandLineAndCheck(base::CommandLine* cmdline,
+                                         const char* switch_name,
+                                         GURL* config_value) {
   if (cmdline->HasSwitch(switch_name)) {
     GURL url = GURL(cmdline->GetSwitchValueASCII(switch_name));
     if (url.is_valid()) {
       *config_value = url;
     }
   }
+#if defined(OFFICIAL_BUILD)
+  CHECK(config_value->is_valid() && config_value->SchemeIsHTTPOrHTTPS());
```

Comment:
why would we e getting this from the command line in an official build?

Reply:
in any case we want to CHECK that the url is valid regardless of whether it came from the command line or gn so I don't think this is the right place to check it. It would actually be better to `static_assert` the value from gn so we know it's broken at build time